File: Sort_Algorithm/Insertion_Sort (삽입정렬)/insertion_Sort.java
package _insertion_sort;
import java.util.*;

public class insertion_Sort {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int arr[] = new int[N];
				
		for(int i=0 ; i<N ; i++) 
			arr[i] = sc.nextInt();
		
		insertion_Sort_ASCE(arr);
		for(int j=0 ; j<N ; j++) 
			System.out.print(arr[j] + " ");
		
		System.out.println();
		
		insertion_Sort_DESC(arr);
		for(int j=0 ; j<N ; j++) 
			System.out.print(arr[j] + " ");
		
	}
	// 오름차순
	public static int[] insertion_Sort_ASCE(int[] arr) {
		for(int i=1 ; i<arr.length ; i++) {
				int index = i;
				int save = arr[i];
				
				for(int j=i-1 ; j>=0 ; j--){
					if(save<arr[j]) {
						arr[index] = arr[j]; 
						index--;
					}
				}
			arr[index] = save;	
		}
		return arr;
	} 
	// 내림차순
	public static int[] insertion_Sort_DESC(int[] arr) {
		for(int i=1 ; i<arr.length ; i++) {
				int index = i;
				int save = arr[i];
				
				for(int j=i-1 ; j>=0 ; j--){
					if(save>arr[j]) {
						arr[index] = arr[j]; 
						index--;
					}
				}
			arr[index] = save;	
		}
		return arr;
	}
}

File: Sort_Algorithm/Insertion_Sort (삽입정렬)/insertion_Sort.swift
func insertion_Sort_ASCE(arr: [Int]) -> [Int] {
	var sortArr = arr
    
    	for i in 1..<sortArr.count {
    		var index = i
        
       		while index > 0 , sortArr[index] < sortArr[index - 1] {
        		sortArr.swapAt(index, index - 1)
            		index -= 1
        	}
    	}
    	return sortArr
}

func insertion_Sort_DESC(arr: [Int]) -> [Int] {
	var sortArr = arr
    
    	for i in 1..<sortArr.count {
    		var index = i
        
       		while index > 0 , sortArr[index] > sortArr[index - 1] {
        		sortArr.swapAt(index, index - 1)
            		index -= 1
        	}
    	}
    	return sortArr
}

var a = [2, 4, 3, 5, 1]

print(insertion_Sort_ASCE(arr: a))
print(insertion_Sort_DESC(arr: a))